syncmap.go

```diff
@@ -0,0 +1,171 @@
+package dapper
+
+import (
+	"io"
+	"reflect"
+	"sort"
+	"strings"
+
+	"github.com/dogmatiq/iago/indent"
+	"github.com/dogmatiq/iago/must"
+)
+
+func mapFilter(
+	w io.Writer,
+	v Value,
+	f func(w io.Writer, v Value) error,
+) (err error) {
+	defer must.Recover(&err)
+
+	if v.IsAmbiguousType() {
+		must.WriteString(w, v.TypeName())
+	}
+	if v.Value.IsZero() {
+		must.WriteString(w, "{}")
+		return
+	}
+
+	i := syncMapItems{}
+	if v.Value.Addr().MethodByName("Range").Call(
```

Comment:
It still feels kind of clunky to use reflection here, I know we talked about it.. but was there some particular reason not to use `Addr().Interface().(*sync.Map)`?